src/main/java/io/vertx5/core/net/NetSocket.java

```diff
@@ -97,7 +114,16 @@ public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
       channelHandlerContext = null;
     }
     @Override
+    public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
+      readInProgress = false;
+      if (needsFlush) {
+        needsFlush = false;
+        ctx.flush();
```

Comment:
I would check, after the flush, if the channel is writable: if not, I would setAutoRead = false and register for the channelWritabilityChange before enabling it to true again.
In this way, if client cannot keep up with reading data (or other network related reasons), netty won't keep on ingesting requests which responses won't be sent back, risking to OOM the server

Reply:
I will do that, that being said there is no OOM risk as user should register its own hook to start/stop writing in the vertx layer.